code/cloud9/cloud9_instance_stack.yaml

```diff
@@ -0,0 +1,364 @@
+---
+AWSTemplateFormatVersion: "2010-09-09"
+Description:
+  AWS CloudFormation template for dynamic Cloud 9 setups. Creates a Cloud9
+  bootstraps the instance.
+Parameters:
+  BucketName:
+    Type: String
+    Default: "cfn-workshop-01-123456789 "
+  C9InstanceType:
+    Description: Cloud9 instance type
+    Type: String
+    Default: t3.small
+    AllowedValues:
+      - t2.micro
+      - t3.micro
+      - t3.small
+      - t3.medium
+    ConstraintDescription: Must be a valid Cloud9 instance type
+  C9EnvType:
+    Description: Environment type.
+    Default: self
+    Type: String
+    AllowedValues:
+      - self
+      - 3rdParty
+    ConstraintDescription: must specify self or 3rdParty.
+  OwnerArn:
+    Type: String
+    Description: The Arn of the Cloud9 Owner to be set if 3rdParty deployment.
+    Default: ""
+  C9InstanceVolumeSize:
+    Type: Number
+    Description: The Size in GB of the Cloud9 Instance Volume.
+    Default: 15
+  Project:
+    Type: String
+    Description: Project Name.
+    Default: ""
+  ShutdownTimeout:
+    Type: Number
+    Description: Inactivity time before shutting down the instance in seconds
+    Default: 3600
+Conditions:
+  Create3rdPartyResources: !Equals [!Ref C9EnvType, 3rdParty]
```

Comment:
See my previous comment if we need to set this

Reply:
> See my previous comment if we need to set this

removed this